assets/src/edit-story/elements/shared/useInsertElement.js

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+  DEFAULT_EDITOR_PAGE_WIDTH,
+  DEFAULT_EDITOR_PAGE_HEIGHT,
+} from '../../constants';
+import { createNewElement, getDefinitionForType } from '../../elements';
+import { editorToDataX, editorToDataY } from '../../units';
+import { useStory } from '../../app';
+
+function useInsertElement() {
+  const {
+    actions: { addElement, setBackgroundElement },
+    state: { currentPage },
+  } = useStory();
+
+  const isMediaEl = (type) => {
+    const { isMedia } = getDefinitionForType(type);
+    return isMedia;
+  };
+  const insertElement = (type, { width, height, isBackground, ...props }) => {
+    const element = createNewElement(type, {
+      ...props,
+      x: editorToDataX(80 * Math.random(), DEFAULT_EDITOR_PAGE_WIDTH),
+      isBackground,
```

Comment:
`setBackgroundElement` already sets `isBackground`.

Reply:
It the background is false, make sure it is set to false.

Reply:
As [per above](https://github.com/google/web-stories-wp/pull/376#discussion_r384631872), treat `isBackground` as read-only.

Reply:
Removed https://github.com/google/web-stories-wp/pull/376/commits/10e7278acaf796fa565433cd06951c5134d9dff5